### ISSUE TITLE
Building with -DENABLE_FILTER_TESTING=ON fails.

### ISSUE THREAD

**Issue by ArchangeGabriel** [Unidata/netcdf-c#810]
Same as in #808, but adding the title flag. At the end of configuration checking:
```
CMake Error at examples/C/hdf5plugins/CMakeLists.txt:18 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "bzip2" because another target with the
  same name already exists.  The existing target is a module library created
  in source directory
  "/build/netcdf/src/netcdf-c-4.6.0/nc_test4/hdf5plugins".  See documentation
  for policy CMP0002 for more details.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5_hl.so" to target "bzip2"
  which is not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5.so" to target "bzip2" which
  is not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libsz.so" to target "bzip2" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libz.so" to target "bzip2" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libdl.so" to target "bzip2" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:22 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libm.so" to target "bzip2" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:24 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "misc" because another target with the
  same name already exists.  The existing target is a module library created
  in source directory
  "/build/netcdf/src/netcdf-c-4.6.0/nc_test4/hdf5plugins".  See documentation
  for policy CMP0002 for more details.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5_hl.so" to target "misc" which
  is not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libsz.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libz.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libdl.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libm.so" to target "misc" which is
  not built in this directory.



```

**Comment by DennisHeimbigner** [comment]
>ADD_LIBRARY cannot create target "bzip2" because another target with the
>same name already exists.  The existing target is a module library created
>  in source directory
I wonder why we are not seeing this error. What version of cmake are you using?

**Comment by ArchangeGabriel** [comment]
3.10.2

**Comment by DennisHeimbigner** [comment]
It appears that we need to use a logical target name to avoid this.
I will experiment with this.

**Comment by DennisHeimbigner** [comment]
I am attaching a modified version of nc_test4/hdf5plugins/CMakeLists.txt.
Can you try it and see if it solves the bug you observe?
[CMakeLists.txt](https://github.com/Unidata/netcdf-c/files/1665935/CMakeLists.txt)

**Comment by ArchangeGabriel** [comment]
OK, I will. Just have something else to finish first.

**Comment by ArchangeGabriel** [comment]
Seems better:
```
CMake Error at examples/C/hdf5plugins/CMakeLists.txt:24 (ADD_LIBRARY):
  ADD_LIBRARY cannot create target "misc" because another target with the
  same name already exists.  The existing target is a module library created
  in source directory
  "/build/netcdf/src/netcdf-c-4.6.0/nc_test4/hdf5plugins".  See documentation
  for policy CMP0002 for more details.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5_hl.so" to target "misc" which
  is not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libhdf5.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libsz.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libz.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libdl.so" to target "misc" which is
  not built in this directory.


CMake Error at examples/C/hdf5plugins/CMakeLists.txt:28 (TARGET_LINK_LIBRARIES):
  Attempt to add link library "/usr/lib/libm.so" to target "misc" which is
  not built in this directory.




```

**Comment by ArchangeGabriel** [comment]
The remaining error were fixed at some point apparently (it works in 4.6.1), closing.
